File: java/src/main/java/com/epam/eng/pattern/strategy/Calculator.java
package com.epam.eng.pattern.strategy;

public class Calculator {

    public static double calculate(double num1, double num2, String operation) {
        if(operation.equals("add")) {
            return num1 + num2;
        } else if(operation.equals("subtract")) {
            return num1 - num2;
        } else if(operation.equals("multiply")) {
            return num1 * num2;
        } else if(operation.equals("divide")) {
            return num1 / num2;
        } else {
            throw new IllegalArgumentException("Unknown operation: " + operation);
        }
    }

    public static void main(String[] args) {
        System.out.println("3 + 4 = " + calculate(3, 4, "add"));
        System.out.println("3 - 4 = " + calculate(3, 4, "subtract"));
        System.out.println("3 * 4 = " + calculate(3, 4, "multiply"));
        System.out.println("3 / 4 = " + calculate(3, 4, "divide"));
    }
}

// todo: consider to abstract algorithm

File: ts/strategy/Calculator.ts
interface CalculationStrategy {
    execute(num1: number, num2: number): number;
}

class Add implements CalculationStrategy {
    execute(num1: number, num2: number): number {
        return num1 + num2;
    }
}

class Subtract implements CalculationStrategy {
    execute(num1: number, num2: number): number {
        return num1 - num2;
    }
}

class Multiply implements CalculationStrategy {
    execute(num1: number, num2: number): number {
        return num1 * num2;
    }
}

class Divide implements CalculationStrategy {
    execute(num1: number, num2: number): number {
        return num1 / num2;
    }
}

class Calculator {
    static calculate(num1: number, num2: number, strategy: CalculationStrategy): number {
        return strategy.execute(num1, num2);
    }
}

console.log("3 + 4 = " + Calculator.calculate(3, 4, new Add()));
console.log("3 - 4 = " + Calculator.calculate(3, 4, new Subtract()));
console.log("3 * 4 = " + Calculator.calculate(3, 4, new Multiply()));
console.log("3 / 4 = " + Calculator.calculate(3, 4, new Divide()));
